doc/reference/guidelines.rst

```diff
@@ -163,7 +163,8 @@ To declare a record in XML, the **record** notation (using *<record>*) is recomm
 - Try to group the record by model. In case of dependencies between
   action/menu/views, this convention may not be applicable.
 - Use naming convention defined at the next point
-- The tag *<data>* is only used to set not-updatable data with ``noupdate=1``
```

Comment:
You are right @mart-e 
I just updated this guideline using the comment from commit message https://github.com/odoo/odoo/commit/7c0e598c97cc48d8e55688c99db6d11ee23d9e46
`The tag <data> is now optional and should only be used if the same xml file contains both update and noupdate records.`